src/lib/uTop.ml

```diff
@@ -32,8 +38,29 @@ let version = "%%VERSION%%"
    | History                                                         |
    +-----------------------------------------------------------------+ *)
 
+(** create a directory but doesn't raise an exception if the directory already exist *)
+let mkdir_safe dir perm =
+    try Unix.mkdir dir perm with Unix.Unix_error (Unix.EEXIST, _, _) -> ()
+
+(** create a directory, and create parent if doesn't exist *)
+let mkdir dir perm =
+    let rec p_mkdir dir =
+        let p_name = Filename.dirname dir in
+        if p_name <> "/" && p_name <> "."
+    then p_mkdir p_name;
```

Comment:
Indentation is misleading here. In general, I would suggest to move to OCamlformat to avoid discussions around indentation.